rust/connlib/ip-packet/src/lib.rs

```diff
@@ -199,7 +199,7 @@ impl std::fmt::Debug for IpPacket {
 }
 
 impl IpPacket {
-    pub fn new(buf: IpPacketBuf, len: usize) -> Result<Self> {
+    pub fn new(buf: IpPacketBuf, len: usize) -> Result<Result<Self, Fragmented>> {
```

Comment:
I wasn't quite sure what the best API here is. Instead of a double-result, we could also return `Fragmented` inside the `anyhow::Error`. That however would be less type-safe because the compiler cannot tell us all the call-sites of this API. Plus, we'd have to get the `Fragmented` error out of the `anyhow::Error` via downcasting.

The double-result isn't pretty but at least it gives us type-safety that we are handling this in all circumstances.